content/classes-part-2/reading/custom-fields/_index.md

```diff
@@ -0,0 +1,251 @@
+draft: false
+weight: 1
+originalAuthor: Courtney Frey # to be set by page creator
+originalAuthorGitHub: speudusa # to be set by page creator
+reviewer: # to be set by the page reviewer
+reviewerGitHub: # to be set by the page reviewer
+lastEditor: # update any time edits are made after review
+lastEditorGitHub: # update any time edits are made after review
+lastMod: # UPDATE ANY TIME CHANGES ARE MADE
+---
+
+In the previous chapter, we used fields to store data within a class, and we explored how to access and modify the values of those fields.
+
+Now, we will explore several ways to configure fields based on their intended use.
+
+## Readonly Fields
+
+A **readonly field** is one that cannot be changed once it is initialized. This means slightly different things for value and reference types. We create readonly fields by declaring them with the `readonly` keyword.
+
+We cannot change the value of a `readonly` value field (`readonly` int, `readonly` double, etc.) after it is initialized.
+
+Similarly, we cannot assign a new object to a `readonly` reference field (`readonly` `ClassName`, for example) after initialization. However, because objects are reference types and not value types, we can change the values within the object itself.
+
+Here are some examples to illustrate. Each class would normally be in its own file, but we present them side-by-side for convenience. Additionally, we declare each field `public` to minimize the example code and more clearly demonstrate where compiler errors would occur.
+
+{{% notice blue "Example" "rocket" %}} 
+
+In addition to Program.cs, we have two classes in a FinalFields project, FinalFields and FortyTwo.
+
+`FortyTwo` contains one field:
+
+   ```csharp{linenos=table,hl_lines=[],linenostart=6}
+   public int intValue = 42;
+   ```
+
+`FinalFields` contains three fields:
+   ```csharp{linenos=table,hl_lines=[],linenostart=6}
+   public readonly int intValue = 42;
+   public readonly double doubleValue;
+   public readonly FortyTwo objectValue = new FortyTwo
+   ```
+
+Let’s see what happens when we try to reassign values to our `readonly` fields in `Program.cs.`
+
+```csharp{linenos=table,hl_lines=[],linenostart=8}
+FinalFields demo = new FinalFields();
+
+// This would result in a compiler error because IntValue has already been initialized.
+demo.intValue = 6;
+
+// This isn't allowed since we didn't initialize DoubleValue in the class declaration.
+demo.doubleValue = 42.0;
+
+// This would result in a compiler error.
+demo.doubleValue = 6.0;
+
+// This would result in a compiler error, since we're trying to
+// give objectValue a different object value.
+demo.objectValue = new FortyTwo();
+
+// However, this is allowed since we're changing a field
+// inside the final object, and not changing which object
+// objectValue refers to.
+demo.objectValue.intValue = 6;
+```
+
+You can test this our in your `csharp-web-dev-examples` repo.  This example is in the Classes-Part-2 solution.
```

Comment:
I found this hidden at the bottom, was thinking move it to the top in a note or tip?  Thoughts?

Reply:
done.  i introduced the location of the example code for each example.